### PR TITLE
leaderboard init

### LINKED ISSUE #127: Contributor leaderboard
It has been discussed across various issues, PRs and discussion threads and with recruitment on-going it might be a good thing for newbies to visually see what other contributors have earned, even 1x'ers like me 🤣

<hr/>

1. List contributors by earnings (only available marker right now)
2. Display contributor's username in place of their EOA
3. display a preview modal with additional metadata about the contributor
4. Later the XP system, Karma, etc... can all be built into the metadata modal.
5. Fetch the leaderboard data from an up to date source (new task for a GH action to run every 24hrs and update the source?)
6. Cache what data we can

### PR DESCRIPTION
**Pull request by Keyrxng**
Resolves https://github.com/ubiquity/.github/issues/127

Requires https://github.com/ubiquity/.github/issues/109

It's still pulling from an old static repo file for the data at the moment.

You need to log in because of the above but once addressed there shouldn't be any need.

### CODE REVIEW

**Comment by 0x4007** [comment]
Build failed. Heads up:

image

File: build/esbuild-build.ts [hunk 1]
```diff
@@ -1,33 +1,77 @@
...
+    ".png": "file",
+    ".woff": "file",
+    ".woff2": "file",
+    ".eot": "file",
+    ".ttf": "file",
+    ".svg": "file",
+    ".json": "file",
   },
   outdir: "static/dist",
+  define: createEnvDefines(["SUPABASE_URL", "SUPABASE_ANON_KEY"], {
```
Comment (0x4007) [inline build/esbuild-build.ts]: This threw the error.

Comment (Keyrxng) [reply build/esbuild-build.ts]: I didn't know the format had changed. Is this change across all repos?

**Comment by zugdev** [comment]
Did this stagnate? If it's because of env I could give you a little push as it seems you are really close to completion.

**Comment by 0x4007** [comment]
> Did this stagnate? If it's because of env I could give you a little push as it seems you are really close to completion.

Good idea to take projects over if they stop yes

**Comment by Keyrxng** [comment]
This is held back because the DB does not have all of the permits in order to generate a leaderboard which was the agreed upon approach as opposed to a cron job that would populate a repo json.

I have had a PR awaiting review for months that is the requirement for this to progress https://github.com/ubiquity/.github/pull/112. I seeded my db with a backup of the official instance although this was months ago when I saw it through to completion. 

Afaik the permit storage is broken and has been for months so newer permits may not even be getting populated right now. I can't find the comment but @gentlementlegen you mentioned it before I'm certain.

> Did this stagnate? If it's because of env I could give you a little push as it seems you are really close to completion.

Please feel free to review https://github.com/ubiquity/.github/pull/112 so that this can move forward if you like zug or get involved in improving the refactoring as it has a lot of weak points Dune being a blatant one ofc, happy to dual-assign and collab if need be or whatever, thanks.

**Comment by gentlementlegen** [comment]
@Keyrxng yes there is a ticket on that regard: https://github.com/ubiquity-os-marketplace/text-conversation-rewards/issues/152

**Comment by Keyrxng** [comment]
I could merge this and just use the static data that I was told not to use previously, would that make sense in terms of being able to close this out or should we wait until https://github.com/ubiquity/.github/pull/112? @0x4007 @rndquu

**Comment by zugdev** [comment]
> I could merge this and just use the static data that I was told not to use previously, would that make sense in terms of being able to close this out or should we wait until https://github.com/ubiquity/.github/pull/112? @0x4007 @rndquu

I understood that, I believe you should wait will need to redo it anyway if you push.

**Comment by Keyrxng** [comment]
The consensus is typically not to wait and to push forward if possible, this has been stagnant for a long time and no clear sign that the required PR will be merged anytime soon so I'm offering solutions.

If using the static data initially is acceptable as I first implemented, I can pull a fresh leaderboard and push this forward. Happy to do that given the green light to get things moving again.

**Comment by rndquu** [comment]
@Keyrxng Pls check https://github.com/ubiquity/.github/issues/127. If current PR is somewhat near the proposed solution then it makes sense to reopen it.

**Pull request closed without merging**